app/database_etl/postgres_tables_handler/table_generator.py

```diff
@@ -203,6 +204,24 @@ def create_country_df(dashboard_source_df, current_time):
     country_df = add_latlng_to_df("country", "country_name", country_df)
     country_df['country_iso3'] = country_df["country_name"].map(lambda a: get_country_code(a))
 
+    # Populate HRP status column
+    hrp_csv_path = os.path.dirname(os.path.abspath(__file__)) + '/country_hrp_map.csv'
+    hrp_countries = set(pd.read_csv(hrp_csv_path)['country_iso_code'])
+    country_df['hrp_class'] = country_df['country_iso3'].map(lambda a: 'HRP' if a in hrp_countries else 'non-HRP')
+
+    # Populate income status column
+    income_csv_path = os.path.dirname(os.path.abspath(__file__)) + '/country_income_class_map.csv'
+    income_class_df = pd.read_csv(income_csv_path)
+    # defining as a closure so that we have access to income_class_df
```

Comment:
What is best practice in a case like this -- defining as a closure, or passing in the mapping df? Asking to improve my understanding of best practices.

Reply:
I think using a closure may be slightly more efficient here (though probably not noticeable) because we wouldn't have to pass the df around in memory so much.

I would consider not using a closure if the function in question could be called in more than 1 context, but that doesn't seem like it will be the case here.

@simonarocco thoughts?

Reply:
I mentioned this in my other PR review -- I personally don't like the pattern of using closures. I think it makes the code less readable and if we start arbitrarily defining some functions as closures while others not as closures (i.e. if diff people write the code with different styles) it can create inconsistencies which will make it harder to maintain. I don't think we will see any noticeable effects of passing the df around.